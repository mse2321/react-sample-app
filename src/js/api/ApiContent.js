import React, { Component } from 'react';
import axios from 'axios';
import CountryTableRow from '../components/CountryTableRow';

class ApiContent extends React.Component {
  constructor(props) {
    super(props)
    this.state = "";
  }

  componentDidMount() {
    axios.get('https://restcountries.eu/rest/v2/all')
      .then(res => {
        const countryData = res.data;
        const numberOfCountries = countryData.length;
        var countryNames = [];

        for(var i = 0; i < numberOfCountries; i++){
          const numberCheck = typeof i;

          if(numberCheck == "number" && i < 251){
            countryNames.push( countryData[i].name.toString() );
            const countryNameList = countryNames.map((countryName) => <CountryTableRow key={ countryName } country_name={ countryName } />);
            this.setState({ countryNameList })
          }
        }

      })
      .catch(error=>{console.log('Error: '+error)})
  }

  render() {
    return (
      <section className="apiContent">
        <table>
          <tr className="HeadingRow">
            <th>Here is a list of countries:</th>
          </tr>
          { this.state.countryNameList }
        </table>
      </section>
    )
  }
}

export default ApiContent;
