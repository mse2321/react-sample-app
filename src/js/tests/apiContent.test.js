import React from 'react';
import ReactDOM from 'react-dom';
import ApiContent from '../api/apiContent';
import renderer from "react-test-renderer";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";

describe('Component: apiContent', () => {

  it('should render a section tag with a list of countries inside', () => {
    const testApiContent = renderer.create(
      <ApiContent />
    ).toJSON();

    expect(testApiContent).toMatchSnapshot();
  });
});
