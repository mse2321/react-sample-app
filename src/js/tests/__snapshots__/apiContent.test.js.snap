// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: apiContent should match the apiContent component 1`] = `
<section
  className="apiContent"
>
  <ul>
    <h2>
      Here is a list of random countries:
    </h2>
     
  </ul>
</section>
`;
